File: ga/4omini_result_2024_7/result/1968C_cpp_mini/3_1968C.cpp
use std::io;

const N: usize = 500;
const A: usize = 1000000000;

fn proc() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let n: usize = input.trim().parse().expect("Failed to parse");

    let mut x = vec![0; N];
    let mut a = vec![0; N];

    input.clear();
    io::stdin().read_line(&mut input).expect("Failed to read line");

    let values: Vec<usize> = input.trim().split_whitespace()
                                   .map(|s| s.parse().expect("Failed to parse"))
                                   .collect();

    for i in 1..n {
        x[i] = values[i - 1]; // Reading values into x
    }

    a[0] = x[1] + 1; // equivalent to a[0] = x[1] + 1 in C++

    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];
            if (if i + 1 < n { x[i + 1] } else { 1 }) < a[i] && a[i] != a[i - 1] {
                break;
            }
        }
    }

    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!(); // Use println! for new line after printing the array.
}

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Failed to read line");
    let t: usize = input.trim().parse().expect("Failed to parse");

    for _ in 0..t {
        proc();
    }
}

File: ga/gpt3_5_result_2024_6/result/1968C_cpp_gpt/3_1968C.rs
use std::io;

const N: usize = 500;
const X: usize = 500;
const A: usize = 1000000000;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let t: usize = input.trim().parse().unwrap();

    for _ in 0..t {
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let n: usize = input.trim().parse().unwrap();

        let mut x: Vec<usize> = vec![0; n];
        let mut a: Vec<usize> = vec![0; n];

        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let mut iter = input.split_whitespace();

        for i in 1..n {
            x[i] = iter.next().unwrap().parse().unwrap();
        }

        a[0] = x[1] + 1;
        for i in 1..n {
            let mut j = 0;
            while j <= A {
                a[i] = x[i] + j * a[i - 1];
                if (i + 1 < n && x[i + 1] < a[i] && a[i] != a[i - 1]) || i + 1 == n {
                    break;
                }
                j += 1;
            }
        }

        for i in 0..n {
            print!("{} ", a[i]);
        }
        println!();
    }
}